bundles/org.smarthomej.binding.viessmann/src/main/java/org/smarthomej/binding/viessmann/internal/dto/features/ViessmannFeatureMap.java

```diff
@@ -0,0 +1,364 @@
+        DESCRIPTION_MAP.put("heating.power.consumption.heating",
+                "Shows statistics of power usage for heating up rooms.");
+        DESCRIPTION_MAP.put("heating.gas.consumption.total", "Shows statistics of gas usage");
+        DESCRIPTION_MAP.put("heating.dhw.temperature.hygiene",
+                "To kill Legionella bacteria the system needs to heat up to at least 65 °C");
+        DESCRIPTION_MAP.put("ventilation.operating.modes.active",
+                "Shows current active operating mode enabled on the device.");
+        DESCRIPTION_MAP.put("ventilation.operating.modes.standard",
+                "Shows whether Standard operating mode is active with minimum air flow.");
+        DESCRIPTION_MAP.put("ventilation.operating.modes.standby", "Shows whether Standby operating mode is active.");
+        DESCRIPTION_MAP.put("ventilation.operating.modes.ventilation",
+                "Shows whether Ventilation operating mode is active.");
+        DESCRIPTION_MAP.put("ventilation.operating.programs.active",
+                "Shows current active operating program enabled on the device.");
+        DESCRIPTION_MAP.put("ventilation.operating.programs.basic",
+                "Shows whether ventilation operating program is active with minimum air flow.");
+        DESCRIPTION_MAP.put("ventilation.operating.programs.intensive",
+                "Shows whether ventilation operating program is active with maximum air flow.");
+        DESCRIPTION_MAP.put("ventilation.operating.programs.reduced",
+                "Shows whether ventilation operating program is active with reduced air flow.");
+        DESCRIPTION_MAP.put("ventilation.operating.programs.standard",
+                "Shows whether ventilation operating program is active with standard air flow.");
+        DESCRIPTION_MAP.put("ventilation.operating.programs.standby",
+                "Shows whether ventilation operating program is in standby.");
+        DESCRIPTION_MAP.put("ventilation.schedule",
+                "Shows schedule of ventilation and provides command to set a new one.");
+        return DESCRIPTION_MAP;
+    }
+
+    public @Nullable String getName(String feature) {
+        getNameMap();
+        Pattern pattern = Pattern.compile("(.[0-3])");
+        Matcher matcher = pattern.matcher(feature);
+        if (matcher.find()) {
+            String circuit = matcher.group(0);
+            feature = matcher.replaceAll(".N");
+            String name = NAME_MAP.get(feature) + " (Circuit: " + circuit.replace(".", "") + ")";
+            return name;
+        }
+        if (NAME_MAP.get(feature) == null) {
+            return feature;
+        }
+        return NAME_MAP.get(feature);
+    }
+
+    public @Nullable String getDescription(String feature) {
+        getDescriptionMap();
+        Pattern pattern = Pattern.compile("(.[0-3])", Pattern.CASE_INSENSITIVE);
```

Comment:
```suggestion
        Pattern pattern = Pattern.compile("(\\.[0-3])", Pattern.CASE_INSENSITIVE);
```

I guess you want to match a dot followed by 0-3", not "any character followed by 0-3" (also above).

Wouldn't `feature.replaceAll("\\.[0-3]", ".N");` also work? If you compile the pattern each time, there is no benefit in using the matcher.

Reply:
> `feature.replaceAll("\\.[0-3]", ".N");`

Yes it works. I have changed it